test/cli_test.py

```diff
@@ -340,35 +342,44 @@ def mock_get_pty_info(shell: bool) -> api_pb2.PTYInfo:
             env_term_program=os.environ.get("TERM_PROGRAM"),
         )
 
+    captured_out = []
+
+    async def write_to_fd(fd: int, data: bytes):
+        nonlocal captured_out
+        captured_out.append((fd, data))
+
     with mock.patch("rich.console.Console.is_terminal", True), mock.patch(
         "modal._pty.get_pty_info", mock_get_pty_info
-    ), mock.patch("modal._pty.write_stdin_to_pty_stream", asyncnullcontext):
-        yield
+    ), mock.patch("modal._container_exec.get_pty_info", mock_get_pty_info), mock.patch(
+        "modal._pty.write_stdin_to_pty_stream", asyncnullcontext
+    ), mock.patch(
+        "modal._container_exec.handle_exec_input", asyncnullcontext
+    ), mock.patch(
+        "modal._container_exec._write_to_fd", write_to_fd
+    ):
+        yield captured_out
 
 
 @pytest.mark.usefixtures("mock_shell_pty")
-def test_shell(servicer, set_env_client, test_dir):
+@skip_windows("modal shell is not supported on Windows.")
+def test_shell(servicer, set_env_client, test_dir, mock_shell_pty):
```

Comment:
Can remove the `mark.usefixtures` if we're also explicitly adding it as an argument